src/web/server/document.tsx

```diff
@@ -176,6 +176,7 @@ export const document = ({ data }: Props): string => {
 			CAPI.config && { src: CAPI.config.commercialBundleUrl },
 			...getByChunkName('sentryLoader'),
 			...getByChunkName('dynamicImport'),
+			...getByChunkName('coreVitals'),
```

Comment:
should this be high priority or low priority? (genuine question)

Reply:
I guess Gareth mentioned logging immediately so perhaps I'll move it to the top.

Reply:
As I understand it, these scripts block the page, so my take would be we should only have performance monitoring there if it _needs_ to load ahead of everything else. But even then I'm not sure we guarentee that here?

Reply:
Yeah, I'm sure the order is not guaranteed

Reply:
Yeah, the order is insignificant if these are asyncronous actions. But maybe we want to drop `coreVitals` down to:

```
	const lowPriorityScriptTags = generateScriptTags([
		...getScriptArrayFromChunkName('lotame'),
		...getScriptArrayFromChunkName('atomIframe'),
		...getScriptArrayFromChunkName('embedIframe'),
		...getScriptArrayFromChunkName('newsletterEmbedIframe'),
	]);
```

where it will be loaded later in the bottom of the page (not the head, which blocks).

But if we do this it then means this code will execute after the initial page load which might break things here?

Reply:
Ideally we need the script running as soon as possible

Reply:
The comment for high priority scripts is

```
	/**
	 * The highest priority scripts.
	 * These scripts have a considerable impact on site performance.
	 * Only scripts critical to application execution may go in here.
	 * Please talk to the dotcom platform team before adding more.
	 * Scripts will be executed in the order they appear in this array
	 */
```

So we should be *very* cautious when adding to it as we are here.

But I'm unsure how the lib works so I can't say this for sure where it needs to be. Eg. Does this lib need to be loaded first thing to start recording information that it sends later or can it be loaded at a later stage and still capture what it needs? My instinct is that it can come later but I think we should run some tests for this.

Reply:
Could we maybe?


```typescript
// App.tsx
import {getCLS, getFID, getLCP} from 'web-vitals';

const reportStuff = () => {}

useOnce(() => {
  getCLS(reportStuff);
  getFID(reportStuff);
  getLCP(reportStuff);
});
```

No idea if that's valid but might be worth a test

Reply:
We can test that impact with this https://github.com/guardian/dotcom-rendering/pull/2559 when the tables are re-created